### ISSUE TITLE
aria-sort on table headers requires a bunch of redundant roles

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/html#909]
There apparently was a [bug filed against the validator](http://lists.w3.org/Archives/Public/www-validator-cvs/2014May/0001.html) that it complained when `aria-sort` is added to a `<th>`. It required `role="columnheader"`  The bug was closed because the spec apparently requires it.
Apparently this is because `<th>` is not supposed to have any semantic value because there are so many layout tables in the wild so the dev has to add a redundant aria role.

It seems like a weird situation, I think it should be perfectly fine to do this and not require redundant aria `<th aria-sort="ascending" ...> `

Can we solve this, and not burden developers by forcing them to add redundant roles?

[Here's a stack overflow discussion](http://stackoverflow.com/questions/24863531/using-aria-sort-in-validated-html5)

**Comment by stevefaulkner** [comment]
@DavidMacDonald this is the wrong spec to file this issue on, also it is not the case that `th` has no semantics - see the API mappings https://w3c.github.io/html-aam/#el-th

**Comment by DavidMacDonald** [comment]
@stevefaulkner 

Then the Validator **should not** throw an error on this, right?
`<th aria-sort="ascending" ...>`
So should the bug be filed there? I've reopened over there.

**Comment by sideshowbarker** [comment]
The *ARIA* spec itself is the relevant spec here.

So, see https://www.w3.org/TR/wai-aria-1.1/#h-aria-sort, which says that `aria-sort` must not be used when the `role` value is anything other than `columnheader` or `rowheader`.

If you want for the ARIA spec to say that `columnheader` or `rowheader` should be allowed even with the explicit `role` value isn’t `columnheader` or `rowheader` then I guess you need to raise an issue against the ARIA spec requesting that.

**Comment by DavidMacDonald** [comment]
@stevefaulkner 
> also it is not the case that th has no semantics - see the API mappings https://w3c.github.io/html-aam/#el-th

I see, it gets  `role="cell"` 

This is very generic. There are two types of cells in HTML tables. Headers `<th>` and data `<td>`. Why not map it to one of those in the API, instead of `cell`?

Then we can file a bug against wai-aria to allow an api mapping of generic `header` in addition to `columnheader` and `rowheader`

this will offload the burden that is on authors currently, which forces them to add an role to `th` every time they want to use `aria-sort`

**Comment by hartman** [comment]
I think it should be noted that [columnheader](https://www.w3.org/TR/wai-aria-1.1/#columnheader) and [rowheader](https://www.w3.org/TR/wai-aria-1.1/#rowheader) themselves are defined as th[scope="col"] and th[scope="row"] base concepts.

The scope attribute itself however is currently a question mark in AAM: https://w3c.github.io/html-aam/#att-scope

**Comment by LJWatson** [comment]
Please note this issue has been closed in relation to the HTML spec. Issues relating to ARIA and the HTML AAM should be filed on the relevant repos:
* [ARIA repo](https://github.com/w3c/aria)
* [HTML AAM](https://github.com/w3c/html-aam)
